### DISCUSSION TITLE
Fetch on Nuxt Bridge

### DISCUSSION THREAD

**Discussion by psycongaroo** [nuxt/bridge#450] [General]
If useFetch and useAsyncData are not avaliable for nuxt 2 bridge, what can we use?
Axios?
Is not avaliable on: [module support](https://modules.nuxtjs.org/?version=2.x-bridge)

**Comment by danielroe** [comment]
[`useLazyAsyncData`](https://v3.nuxtjs.org/docs/usage/data-fetching/#uselazyasyncdata) and [`useLazyFetch`](https://v3.nuxtjs.org/docs/usage/data-fetching/#uselazyfetch) are compatible with Nuxt Bridge.

**Reply by psycongaroo** [reply]
Thank you, what we can use for interceptors?
I should do something like a composable and calling in it useLazyAsyncData or useLazyFetch?

We can use composables on plugins too?

**Reply by some-user123** [reply]
`useLazyAsyncData` is available but is only partially able to replace `useAsyncData`. The `lazy` variants require to deal with pending data and error situations in the template.

I'm wondering, what's the designed migration path to get from `useAsync` in nuxt2 to `useAsyncData` in nuxt3 when taking nuxt-bridge as intermediate step? (cf. https://stackoverflow.com/questions/73165289/how-to-migrate-useasync-error-to-nuxt-bridge-useasyncdata-is-not-available)

**Comment by infovore** [comment]
If they are compatible, does that mean that this piece of documentation:

![image](https://user-images.githubusercontent.com/1103/156146046-5c13f475-771c-49cc-af12-6b2e6e1f9fbe.png)

(from https://v3.nuxtjs.org/getting-started/bridge/)

is incorrect? And if so, should it updated? Or is there a clarification requried?

**Reply by danielroe** [reply]
The point is that the lazy versions of these composables are compatible, but not the non-lazy versions.

**Reply by infovore** [reply]
Aha, understood! I didn't find that immediately clear from the docs. Appreciate the clarification.
